### DISCUSSION TITLE
will it support iOS?

### DISCUSSION THREAD

**Discussion by akimky** [raysan5/raylib#2681] [General]
I would like to make games on this library on iOS, is there or will there be such an opportunity?

**Comment by raysan5** [comment]
@Akimpaus I never had an iOS or a mac so I couldn't ported it. Anyone interested in iOS can port raylib to it.

**Comment by blueloveTH** [comment]
Is there any progress? I am interested about porting raylib to iOS. However, I need some advanced guide of raysan5... I know iOS does not support opengl. Is the porting really feasible?

**Reply by blueloveTH** [reply]
I see this: https://github.com/raysan5/raylib/discussions/2036.
Can [PortableGL](https://github.com/rswinkle/PortableGL) be used as iOS backend?

**Reply by akimky** [reply]
@blueloveTH I'm sure it's possible. I can't say exactly how, because I have no experience in this matter. Ideally, you need to use Metal, not OpenGL. There are all kinds of wrappers and so on. But again, I'm not good at this... I'm glad that someone is interested in this)))

**Reply by blueloveTH** [reply]
Find some useful note here: https://github.com/raysan5/raylib/discussions/2998#discussioncomment-6069177

**Reply by scripturial** [reply]
It is impossible to predict the future, but I honestly don't think Apple will pull OpenGL support for a very long time. It would kill a large number of of legacy apps in the App Store.  I assume they would pull OpenGL support once the majority of apps no longer depend on OpenGL, and it's impossible to predict when that will be.  But with Tim Cooke, anything is possible.

**Comment by jamesl-github** [comment]
https://www.anandtech.com/show/12894/apple-deprecates-opengl-across-all-oses

by Ryan Smith <https://www.anandtech.com/Author/85> *on June 5, 2018 9:30
AM EST*

"Apple never added support for OpenGL ES 3.1 or later on iOS, and similarly
macOS doesn’t go beyond OpenGL 4.1 (Khronos is up to 4.6 now). Instead, Apple
has been pushing developers to Metal
<https://www.anandtech.com/show/9352/apples-metal-api-comes-to-os-x-desktops>
almost as soon as it became available. And now by deprecating support for
these older APIs, Apple is signaling that they are reserving the right to
remove them entirely in the future."

**Reply by gen2brain** [reply]
Who cares if they remove or not OpenGL? For all this time nobody has stepped up to do anything about the iOS or macOS. If users don't care why should anyone pay attention at all?

**Comment by mrylmz** [comment]
I could eventually work on this issue if I can find some free time.

**Reply by raysan5** [reply]
That would be nice! Note that there is an open PR with some implementation...

**Comment by scripturial** [comment]
I am planning on trying to port my iOS app to zig/raylib. It looks like there is enough prior work that it should be possible. I probably won't be able to look at this for a few months though, right now I am still learning zig in my spare time.  

There will always be people that complain that OpenGL is deprecated on iOS. I am a pragmatist, if it works it works. If Apple ever did remove OpenGL support, I will worry about it when it happens. To be honest, I am betting there are enough apps in the App Store using OpenGL that Apple won't just flat out delete OpenGL overnight.

The challenge for me, will be how it would work outside of zig.  `zig build` can easily pull in an objective c .mm file to provide the iOS glue, but I don't know how it would work with other languages.

**Reply by coffeebe4code** [reply]
Do you have any progress @scripturial I have started using raylib-zig/raylib-zig and would like to have an iOS target as well

**Comment by dhombios** [comment]
Maybe something like this could be used: https://github.com/openglonmetal/MGL. However, I haven't check if it provides every method used by raylib

**Comment by paulocoutinhox** [comment]
Add the iOS/tvOS support pls, only left these platforms.

**Comment by ghera** [comment]
I’m currently working on this in a forked repo (https://github.com/ghera/raylib-iOS),
which I’m actively using to develop an Android/iOS game.

You can find:
- an Xcode 26 example project using the latest (12/2025) stable Apple Silicon
  ANGLE build
- a stable `5.5` branch and a `master` branch frequently aligned with upstream
- notes on the very small `main.c` changes required for iOS

For more details, see the PR my work is based on:
https://github.com/raysan5/raylib/pull/3880

**Comment by gameingame-eng** [comment]
am i late guys

**Reply by ghera** [reply]
Not really :)

There’s no official iOS support yet, but it works using ANGLE. I’ve been maintaining a fork and keeping it up to date:
https://github.com/ghera/raylib-iOS

I’ve also shipped a small iOS game using this approach:
https://www.reddit.com/r/raylib/comments/1scdh81/i_just_published_my_raylib_game_patchwall_on_ios/
